packages/esm-appointments-app/src/appointment-forms/follow-up-form.component.tsx

```diff
@@ -0,0 +1,298 @@
+import React, { useState, useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import dayjs from 'dayjs';
+import {
+  Button,
+  ButtonSet,
+  ContentSwitcher,
+  DatePicker,
+  DatePickerInput,
+  RadioButton,
+  RadioButtonGroup,
+  Select,
+  SelectItem,
+  Switch,
+  TextArea,
+  TimePicker,
+  TimePickerSelect,
+  Toggle,
+  TextInput,
+  Layer,
+} from '@carbon/react';
+import {
+  useLocations,
+  useSession,
+  showToast,
+  showNotification,
+  ExtensionSlot,
+  usePatient,
+  useConfig,
+  parseDate,
+} from '@openmrs/esm-framework';
+import { AppointmentPayload, MappedAppointment } from '../types';
+import { amPm } from '../helpers';
+import { saveAppointment, useServices } from './appointment-forms.resource';
+import { ConfigObject } from '../config-schema';
+import { useProviders } from '../hooks/useProviders';
+import { closeOverlay } from '../hooks/useOverlay';
+import { mockFrequency } from '../../__mocks__/appointments.mock';
+import styles from './edit-appointment-form.scss';
+
+interface FollowUpFormProps {
+  appointment: MappedAppointment;
+  mutate: () => void;
+}
+const FollowUpForm: React.FC<FollowUpFormProps> = ({ appointment, mutate = () => {} }) => {
+  const { t } = useTranslation();
+  const { patient } = usePatient(appointment.patientUuid);
+  const session = useSession();
+  const { providers } = useProviders();
+  const { services } = useServices();
+  const [selectedLocation, setSelectedLocation] = useState(appointment.location);
+  const [selectedService, setSelectedService] = useState(appointment.serviceUuid);
+  const [selectedProvider, setSelectedProvider] = useState(session?.currentProvider?.uuid);
+  const [reminder, setReminder] = useState('');
+  const [typeOfTracing, setTypeOfTracing] = useState('');
+  const [tracingOutcome, setTracingOutcome] = useState('');
+  const [finalOutcome, setFinalOutcome] = useState('');
+  const [outComeTrueStatus, setOutComeTrueStatus] = useState('');
+  const [other, setOther] = useState('');
+  const [appointmentComment, setAppointmentComment] = useState(appointment.comments);
+  const [followUpComment, setfollowUpComment] = useState('');
+  const [reason, setReason] = useState('');
+  const [timeFormat, setTimeFormat] = useState<amPm>(new Date().getHours() >= 12 ? 'PM' : 'AM');
+  const [visitDate, setVisitDate] = React.useState<Date>(appointment.dateTime ? parseDate(appointment.dateTime) : null);
+  const [isFullDay, setIsFullDay] = useState<boolean>(false);
+  const [day, setDay] = useState(appointment.dateTime);
+  const [tracingNumbering, setTracingNumbering] = useState('');
+  const [appointmentKind, setAppointmentKind] = useState(appointment.appointmentKind);
+  const [appointmentStatus, setAppointmentStatus] = useState(appointment.status);
+  const [isSubmitting, setIsSubmitting] = useState(false);
```

Comment:
Looks like a good time to leverage a form state management library like react-hook-forms